### PR TITLE
Code for Pop up to select the template for a particular step

### PR DESCRIPTION
**Pull request by chirag-compro**

### CODE REVIEW

File: client/app/task-builder/task-builder.component.ts [hunk 1]
```diff
@@ -46,4 +66,39 @@ initialiseTaskData() {
...
+			this.modalWindow.show();
+		}
+	}
+	setTempalateMap(selectedTemplate){
+		this.SelectedStep.TemplateName=selectedTemplate;
+		this.taskDataService.setTaskTemplate(this.TaskData["id"],this.SelectedStep.Index,selectedTemplate)
+				.subscribe(res =>{
+					this.message =MessageMap[res.message];
+					if(this.message == "Task Template Updated"){
+					Messenger.options = { extraClasses: 'messenger-fixed messenger-on-top',
```
Comment (vsingb3) [inline client/app/task-builder/task-builder.component.ts]: Separate Function

File: server/modules/taskData/BilliTaskData.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 var TaskdataInterface = require('./taskdataInterface'); //all the functions of this class are to be overloaded by the class extending it.
+const  taskTemplateMapModel = require('../../models/taskTemplateMap.model');
```
Comment (vsingb3) [inline server/modules/taskData/BilliTaskData.js]: remove this now

File: server/routes/taskBuilderData.js [hunk 3]
```diff
@@ -20,16 +23,83 @@ router.get("/",function(req,res){
...
+			stepData[stepIndex].setSkillName(skillName);
+			stepData[stepIndex].setTemplateName(skillName);
+			Promise.resolve();			
+		}));	
+	}
+	return Promise.all(promiseArr).then(()=>{
+		return Promise.resolve(taskData)
+	});
+}
+function getSkillName(stepIndex,taskId){
```
Comment (vsingb3) [inline server/routes/taskBuilderData.js]: taskId should be the 1st argument

File: server/routes/taskBuilderData.js [hunk 4]
```diff
@@ -20,16 +23,83 @@ router.get("/",function(req,res){
...
+function mapTaskData(res, modelType){
+var taskModel = taskFactory.getTaskDataModel(modelType, res);
+let taskData = new Task (taskModel);
+return validateTaskData(taskData);
+}
+function validateTaskData(taskData){
+	let promiseArr = [];
+	let stepData = taskData.getStepData();
+	for(let stepIndex=0;stepIndex<stepData.length;stepIndex++){ //function getSkillName to bre renamed to getTemplateName if Baloo gives skill name and set only template name in it..
+			promiseArr.push(getSkillName( stepData[stepIndex].getIndex(),taskData.getId()).then((skillName)=>{
```
Comment (vsingb3) [inline server/routes/taskBuilderData.js]: getSkillName should be getTemplateName

File: server/public/TemplateOptions.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
+    "templateOptions":[
+        {
+            "name":"Move Cell Content",
+            "id":"MoveCellContent"
+        },
+        {
+            "name":"Add New Template!!",
```
Comment (dhirendra-singh81) [inline server/public/TemplateOptions.json]: Add real template ids

**Review by dhirendra-singh81 — APPROVED** [review]

**Pull request merged**